classes/Hook.php

```diff
@@ -1109,6 +1117,23 @@ private static function getHookRegistry()
         return null;
     }
 
+    /**
+     * @return HookModuleFilter|null
+     * @throws \PrestaShop\PrestaShop\Core\Exception\ContainerNotFoundException
+     */
+    private static function getHookModuleFilter()
+    {
+        $serviceContainer = ContainerBuilder::getContainer('front', _PS_MODE_DEV_);
```

Comment:
Hmm, it's surprising that you only fetch the container for front 🤔 The fetched container should adapt on the environment you are currently in, in BO the admin container (Symfony one) in legacy environment (like the FO) the legacy one based on `Adapter\ContainerBuilder`

This code is executed in all environments, and it's at the core of the Hook execution which is used pretty much everywhere, so with the current code it you are forcing the creation and build of the Front container even when we are in the BO which is not needed and could create some unexpected side effects

Reply:
You should use `SymfonyContainer::getInstance` first, and when it returns null it means you are in an FO environment so you could use the legacy COntainerBuilder as a fallback in this case

It also means something very important, the filter service you introduced along with the interfaces implemented in the modules must be present in ALL containers (both legacy and symfony) meaning you shouldn't define them in the front services only but in the common ones https://github.com/PrestaShop/PrestaShop/blob/e8c31d70a6d52ed235a60caf673c98a5db08a029/config/services/common.yml

Finally, you only declared your service in the `prod` which means your feature won't work when the shop is in dev or test mode

Fro what I remember your filter feature should be needed everywhere not only in the FO environment and only in prod, so you need all these adjustments to make your implementation more generic

Reply:
I dealt with those feedbacks 👍  (except about the `prod` config, the dev config actually extends the prod one...)